src/shortcut.js

```diff
@@ -0,0 +1,330 @@
+  Blockly.ShortcutRegistry.registry.addKeyMapping(
+      Blockly.utils.KeyCodes.BACKSPACE, deleteShortcut.name);
+};
+
+const copyData = new Set();
+
+/**
+ * Keyboard shortcut to copy multiple selected blocks on
+ * ctrl+c, cmd+c, or alt+c.
+ */
+const registerCopy = function() {
+  const CopyShortcut = {
+    name: Blockly.ShortcutItems.names.COPY,
+    preconditionFn: function(workspace) {
+      const selected = Blockly.common.getSelected();
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress() &&
+      selected && selected.isDeletable() && selected.isMovable();
+    },
+    callback: function(workspace, e) {
+      // Prevent the default copy behavior, which may beep or
+      // otherwise indicate an error due to the lack of a selection.
+      e.preventDefault();
+      copyData.clear();
+      workspace.hideChaff();
+      const apply = function(block) {
+        if (block && !hasSelectedParent(block)) {
+          copyData.add(block.toCopyData());
+        }
+      };
+      const selected = Blockly.common.getSelected();
+      Blockly.Events.setGroup(true);
+      if (!blockSelection.size) {
+        apply(selected);
+      }
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        apply(block);
+      });
+      Blockly.Events.setGroup(false);
+      return true;
+    },
+  };
+  Blockly.ShortcutRegistry.registry.register(CopyShortcut);
+
+  const ctrlC = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.CTRL]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(
+      ctrlC, Blockly.ShortcutItems.names.COPY);
+
+  const altC =
+  Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.ALT]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(
+      altC, Blockly.ShortcutItems.names.COPY);
+
+  const metaC = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.META]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(
+      metaC, Blockly.ShortcutItems.names.COPY);
+};
+
+
+/**
+ * Keyboard shortcut to copy and delete multiple selected blocks on
+ * ctrl+x, cmd+x, or alt+x.
+ */
+const registerCut = function() {
+  const cutShortcut = {
+    name: Blockly.ShortcutItems.names.CUT,
+    preconditionFn: function(workspace) {
+      const selected = Blockly.common.getSelected();
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress() &&
+      selected && selected.isDeletable() && selected.isMovable() &&
+          !selected.workspace.isFlyout;
+    },
+    callback: function() {
+      copyData.clear();
+      const apply = function(block) {
+        if (block &&
+            block.isDeletable() &&
+            !hasSelectedParent(block)) {
+          copyData.add(block.toCopyData());
+          if (block.workspace.isFlyout) {
```

Comment:
I think this behavior makes sense, but just to check: you want the block to be added to the `copyData` even if it exists in the flyout?